src/uadk_prov_cipher.c

```diff
@@ -634,9 +637,10 @@ static int uadk_prov_do_cipher(struct cipher_priv_ctx *priv, unsigned char *out,
 	int outlint = 0;
 	int ret;
 
-	if (priv->switch_flag == UADK_DO_SOFT ||
-	    (priv->sw_cipher && priv->switch_flag != UADK_DO_HW &&
-	     inlen <= priv->switch_threshold)) {
+	if (priv->sw_cipher &&
+	    (priv->switch_flag == UADK_DO_SOFT ||
+	    (priv->switch_flag != UADK_DO_HW &&
+	     inlen <= priv->switch_threshold))) {
```

Comment:
Maybe add a comment for these three situations will be clearer?

Reply:
sure